apps/wakunode2/wakunode2.nim

```diff
@@ -13,30 +14,32 @@ import
   system/ansi_c,
   libp2p/crypto/crypto
 import
+  ../../waku/v2/waku_core/message/message,
   ../../waku/common/logging,
   ./config,
   ./app
 
 logScope:
   topics = "wakunode main"
 
+const wakuNode2VersionString* = "version / git commit hash: " & git_version
 
-{.pop.} # @TODO confutils.nim(775, 17) Error: can raise an unlisted exception: ref IOError
-when isMainModule:
-  ## Node setup happens in 6 phases:
-  ## 1. Set up storage
-  ## 2. Initialize node
-  ## 3. Mount and initialize configured protocols
-  ## 4. Start node and mounted protocols
-  ## 5. Start monitoring tools and external interfaces
-  ## 6. Setup graceful shutdown hooks
+var wakunode2 {.threadvar.}: App
+var confRes:ConfResult[WakuNodeConf]
 
-  const versionString = "version / git commit hash: " & app.git_version
+{.pop.} # @TODO confutils.nim(775, 17) Error: can raise an unlisted exception: ref IOError
+proc init*(configFilePath = "") =
```

Comment:
I don't think we should `init` an app here, but operate on the `WakuNode` object as defined in `waku_node.nim` (see comments elsewhere). `wakunode2` is simply an application written around the Nim API. As stated elsewhere, we want to provide C bindings wrapping the Nim API, not the `wakunode2` application

Reply:
> I see `init` here uses a path vs the rfc which suggests a json string. We should decide which approach to use to unify behaviors between implementations. cc: @danielSanchezQ for his thoughts as he has worked with the rust+go bindings before

IMO using a configuration is better. A config file should be used for a binary, a configuration object for a library.

@jm-clius Where can I find the nim api documentation? Don't know if Im understanding correctly, but, does it mean that the `wakunode2` is an external application running and the bindings just communicate with it. Or do this works in a similar way to go-waku in where we do embedded the node within the application?

Reply:
The application simply provides a way to configure and instantiate the node, so I guess more similar to "embedded within the application". The Nim API documentation is sparse and outdated (since it's only used internally), but in theory lives here: https://github.com/waku-org/nwaku/blob/master/docs/api/v2/node.md

Reply:
Oh, it's very outdated, in fact. 😟 @Ivansete-status perhaps a good idea that we clean the Nim API doc up at some point during the development of the C bindings, since we're likely to expand/change it in any case.